### DISCUSSION TITLE
How would you use conditional values like $this->when()

### DISCUSSION THREAD

**Discussion by ahoiroman** [spatie/laravel-data#7] [Q&A] [answered]
Hello,

I am just checking out this package, which really seems very interesting.

When it comes to replace Requests and Api-Resources, I am wondering, how I would add conditional values like

```php
'superpowers'      => $this->when(auth()->user()->can('view', $this->resource), $this->superpowers),
```

**Comment by ahoiroman** [comment, marked as answer]
Well - I found it in the docs (the search does not work for me, so I did not find it before asking here).

It can be found here: https://spatie.be/docs/laravel-data/v1/as-a-resource/lazy-properties#conditional-lazy-properties

**Reply by freekmurze** [reply]
Search will work as soon as this one gets merged: https://github.com/algolia/docsearch-configs/pull/4795
